### DISCUSSION TITLE
Suggestions for customizing application help to show messages (e.g. like argparse)?

### DISCUSSION THREAD

**Discussion by brijow** [facebookresearch/hydra#2450] [Q&A] [answered]
The docs for how to customize application help (here: https://hydra.cc/docs/configure_hydra/app_help/) shows how to make some high level customizations to the overall application's help, but I am interested in customizing the help for options within the config at a much more granular level similar to the kind of help associated with options when using argparse.

For example, from the argparse docs, this is the type of help we get for a simple application -- in particular, my focus here is on the help we see associated with the `--sum` option:

```bash
python prog.py -h
usage: prog.py [-h] [--sum] N [N ...]

Process some integers.

positional arguments:
 N           an integer for the accumulator

options:
 -h, --help  show this help message and exit
 --sum       sum the integers (default: find the max
```

I was looking for a way to do something similar with hydra -- that is, creating a custom help message to accompany each option for each config group -- perhaps using Structured Configs as they can be used as a schema (i.e. validate configs to ensure valid values are provided to options), but I couldn't find way of doing so. 

I thought one way may be to create a duplicate set of config files that include a help message (which could be used only when user writes --help), but otherwise work with the standard config files that are "helpless". Alternatively, I suppose could fall back to using argparse and only hydra via Compose API, but I'd like to utilize tab completion and hydra's logging management so that approach 

Was wondering if anyone had a suggestion or link to an example that customizes the help for options as described above?

**Comment by Jasha10** [comment, marked as answer]
Hi @brijow,

I believe that feature request #633 is similar to what you propose.

**Reply by brijow** [reply]
Hi @Jasha10 , yes this is essentially the same request. Thanks for point me to this. 🙏
